FirebaseAuth/Sources/Swift/User/User.swift

```diff
@@ -1746,8 +1753,14 @@ extension User: NSSecureCoding {}
     self.phoneNumber = phoneNumber
     self.metadata = metadata ?? UserMetadata(withCreationDate: nil, lastSignInDate: nil)
     self.tenantID = tenantID
-    // The `heartbeatLogger` and `appCheck` will be set later via a property update.
+    // This property will be overwritten later via the `user.auth` property update. This is to
+    // provide the `heartbeatLogger` and `appCheck` will be set later via the `user.auth` property
+    // update. For now, a placeholder is set as the property update should happen right after this
+    // intializer.
     requestConfiguration = AuthRequestConfiguration(apiKey: apiKey, appID: appID)
+    // This property will be overwritten later via the `user.auth` property update. For now, a
+    // placeholder is set as the property update should happen right after this intializer.
+    backend = AuthBackend(rpcIssuer: AuthBackendRPCIssuer())
```

Comment:
It feels like we need a bigger refactor so that we don't have the same properties on both the Auth and User objects  - but that is a separate project from this PR.

Reply:
Thanks for the review. I agree as well.